x-pack/plugins/security_solution_serverless/public/get_started/hooks/use_check_step_completed.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useRef } from 'react';
+import { useKibana } from '../../common/services';
+import type {
+  StepId,
+  CardId,
+  SectionId,
+  CheckIfStepCompleted,
+  ToggleTaskCompleteStatus,
+} from '../types';
+
+interface Props {
+  autoCheckIfStepCompleted?: CheckIfStepCompleted;
+  cardId: CardId;
+  indicesExist: boolean;
+  sectionId: SectionId;
+  stepId: StepId;
+  toggleTaskCompleteStatus: ToggleTaskCompleteStatus;
+}
+
+export const useCheckStepCompleted = ({
+  autoCheckIfStepCompleted,
+  cardId,
+  indicesExist,
+  sectionId,
+  stepId,
+  toggleTaskCompleteStatus,
+}: Props) => {
+  const kibanaServicesHttp = useKibana().services.http;
+  const abortSignal = useRef(new AbortController());
+
+  useEffect(() => {
+    if (!autoCheckIfStepCompleted) {
+      return;
+    }
+
+    const autoCheckStepCompleted = async () => {
+      const isDone = await autoCheckIfStepCompleted({
+        indicesExist,
+        abortSignal,
+        kibanaServicesHttp,
+      });
```

Comment:
This call may perform API requests, but I can't find any error catching. Can we wrap this call or the specific API requests in try/catch?

Reply:
Implemented in https://github.com/elastic/kibana/pull/171078/commits/09eed5ba5c0bd2849bf38e2a6ffd0dd076bce5f3